gusto/diagnostics.py

```diff
@@ -401,12 +401,47 @@ def name(self):
         return self.field1+"_perturbation"
 
 
+class Precipitation(DiagnosticField):
+    name = "Precipitation"
+
+    def setup(self, state):
+        if not self._initialised:
+            space = state.spaces("DG1", state.mesh, "DG", 1)
+            super(Precipitation, self).setup(state, space=space)
+
+            rain = state.fields('rain')
+            rho = state.fields('rho')
+            v = state.fields('rainfall_velocity')
+            phi = TestFunction(space)
+            flux = TrialFunction(space)
+            n = FacetNormal(state.mesh)
+            un = 0.5 * (dot(v, n) + abs(dot(v, n)))
+            self.flux = Function(space)
+            self.total_water = Function(space)
+
+            if space.extruded:
+                self.ds = ds_b + ds_t + ds_v
+            else:
+                self.ds = ds
+
+            a = phi * flux * dx
+            L = phi * rain * un * rho * self.ds
+
+            # setup solver
+            problem = LinearVariationalProblem(a, L, self.flux)
+            self.solver = LinearVariationalSolver(problem)
+
+    def compute(self, state):
+        self.solver.solve()
+        self.total_water.assign(self.total_water + self.flux)
+        return self.field.interpolate(self.total_water)
+
+
 class Vorticity(DiagnosticField):
     """Base diagnostic field class for vorticity."""
 
     def setup(self, state, vorticity_type=None):
         """Solver for vorticity.
-
```

Comment:
???

Reply:
Ah sorry, it wasn't deliberate. I had copied and pasted this code over from a different branch that had other features and somehow copying into the emacs window has deleted these lines!